File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Services/ServicoAvaliacao.cs
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;
using SistemaAcademico.Validation;

namespace SistemaAcademico.Services;

// Equivalente à classe Java: ServicoAvaliacao
// Injeção de dependência via construtor (Dependency Injection preservado)
public class ServicoAvaliacao
{
    private readonly ServicoTurma servicoTurma;

    public ServicoAvaliacao(ServicoTurma servicoTurma)
    {
        this.servicoTurma = servicoTurma;
    }

    public void RegistrarAvaliacao(string codigoTurma, string nome, string tipo, double valor, double peso, string usuarioLogado)
    {
        if (!"PROFESSOR".Equals(usuarioLogado))
        {
            throw new ExcecaoAutorizacao("Operação negada: Apenas professores podem registrar avaliações.");
        }

        Turma? turmaEncontrada = null;
        foreach (Turma t in servicoTurma.ListarTurmas())
        {
            if (t.Codigo.Equals(codigoTurma))
            {
                turmaEncontrada = t;
                break;
            }
        }

        if (turmaEncontrada == null)
        {
            throw new ExcecaoSistemaAcademico("Turma não encontrada: " + codigoTurma);
        }

        Avaliacao novaAvaliacao = AvaliacaoFactory.Criar(tipo, nome, valor, peso);
        ValidadorDominio.Validate(novaAvaliacao);

        turmaEncontrada.AdicionarAvaliacao(novaAvaliacao);
    }
}

File: academic-system-java/src/main/java/org/example/service/ServicoAvaliacao.java
package org.example.service;

import org.example.model.Turma;
import org.example.model.Avaliacao;
import org.example.model.AvaliacaoFactory;
import org.example.exception.ExcecaoSistemaAcademico;
import org.example.exception.ExcecaoAutorizacao;
import org.example.validation.ValidadorDominio;

public class ServicoAvaliacao {

    private final ServicoTurma servicoTurma;

    public ServicoAvaliacao(ServicoTurma servicoTurma) {
        this.servicoTurma = servicoTurma;
    }

    public void registrarAvaliacao(String codigoTurma, String nome, String tipo, double valor, double peso, String usuarioLogado) {
        if (!"PROFESSOR".equals(usuarioLogado)) {
            throw new ExcecaoAutorizacao("Operação negada: Apenas professores podem registrar avaliações.");
        }

        Turma turmaEncontrada = null;
        for (Turma t : servicoTurma.listarTurmas()) {
            if (t.getCodigo().equals(codigoTurma)) {
                turmaEncontrada = t;
                break;
            }
        }

        if (turmaEncontrada == null) {
            throw new ExcecaoSistemaAcademico("Turma não encontrada: " + codigoTurma);
        }

        Avaliacao novaAvaliacao = AvaliacaoFactory.criar(tipo, nome, valor, peso);
        ValidadorDominio.validate(novaAvaliacao);

        turmaEncontrada.adicionarAvaliacao(novaAvaliacao);
    }
}
